docs/source/miscellaneous/doc/adr/0016-use-tickets-encoding-for-transactions.md

```diff
@@ -253,14 +255,256 @@ performance would be poor for services with many nodes.
+timestamps, the bloom filter we had for the ``_transactions2`` table even with this setting was about 15 KB, while that
+for ``_transactions`` was about 580 MB.
+
 #### Index Intervals
 
+In Cassandra, a partition index is an on-disk file that stores mappings from partition keys to their offset within
+an SSTable. Cassandra maintains a partition summary for each SSTable in memory, which samples the partition index every
+N keys and stores an offset for the location of the mapping for a given key within the file. The value of N may be
+tuned; smaller values of N require more memory, but improve performance as seeks within the partition index will
+be shorter.
+
+For transactions2, the number of partitions is expected to be very small, so we set the ``min_index_interval`` and
+``max_index_interval`` to 1, forcing the partition index to be perfect.
+
 #### Compression Chunk Length
 
+Cassandra compresses SSTables by blocks on disk. These blocks are of configurable size; choosing larger blocks may
+enable better compression (since similarities between columns or values may be exploited) at the expense of needing
+to read more data from disk when a read occurs. Typically in Atlas, this is set to 4 KB to reduce the amount of I/O
+we need to do.
+
+For transactions2, we expect that some users will often be reading data from a relatively smaller working set. In this
+case, using a larger chunk size enables better compression and increases the proportion of said working set that can be
+maintained in memory. We experimented with several settings and found 64 KB to be a good balance.
+
+#### Empirical Evaluation of Cassandra Table Parameters
+
+We ran several benchmarks on an internal testing stack, with different levels of concurrency and different hit rates
+for the transactions table. For these benchmarks, we ensured that we actually performed disk operations to evaluate
+performance. This was done by using ``sstableloader`` to ingest billions of timestamp pairs into the test stack, and
+the ``stress`` tool to consume almost all of the free memory outside of Cassandra's heap (to avoid operating system
+page caches). We also had to create fresh ``TransactionService``s on each benchmark run to circumvent application-level
+caching.
+
+These benchmarks were run with a 100% hit rate (every timestamp queried for corresponded to an existing transaction).
+We attempted to run the tests using various configurations of the optimisations discussed above:
+
+- ``tx1`` is simply the transactions1 algorithm.
+- ``tx2`` is the transactions2 algorithm, using standard AtlasDB table settings (that is, ``bloom_filter_fp_chance`` of
+  0.01, ``min_index_interval`` and ``max_index_interval`` unspecified, and ``chunk_length_kb`` of 4).
+- ``BF`` refers to explicit configuration of ``bloom_filter_fp_chance`` to 0.0001.
+- ``II`` refers to explicit configuration of ``min_index_interval`` and ``max_index_interval`` to be 1.
```

Comment:
lol I definitely wouldn't have guessed.

likely pushing it, but maybe a chart? 😁

Reply:
pushing it... but added 😅

Reply:
amazing